include/nighthawk/adaptive_load/step_controller.h

```diff
@@ -0,0 +1,50 @@
+#pragma once
+
+#include "envoy/common/pure.h"
+#include "envoy/config/core/v3/base.pb.h"
+#include "envoy/config/typed_config.h"
+
+#include "external/envoy/source/common/protobuf/protobuf.h"
+
+#include "api/adaptive_load/benchmark_result.pb.h"
+
+namespace Nighthawk {
+namespace AdaptiveLoad {
+
+// An interface for custom StepControllers that compute load adjustments and check for convergence.
+class StepController {
+public:
+  virtual ~StepController() = default;
+  // Returns the current CommandLineOptions load specification that the StepController recommends.
+  virtual nighthawk::client::CommandLineOptions GetCurrentCommandLineOptions() const PURE;
```

Comment:
Maybe return `const  nighthawk::client::CommandLineOptions&`?

Reply:
Looked into this, but I realized that in my current implementation I'm actually building a clean copy of the CommandLineOptions template on the stack rather than keeping a running CommandLineOptions class member, and it won't let me return a reference to the stack variable.

If we make the API pass the proto around by value, it's less constraining on the StepController implementer, at the cost of making copies of this fairly small proto.

Reply:
Makes sense, thanks for elaborating